### PR TITLE
feat(link-chooser): add accessible scrolling

### PR DESCRIPTION
**Pull request by SamCreasey**
This pr fixes an issue regarding link chooser accessibility. Previously if the link chooser was being navigated via keyboard the dropdown would not scroll to the focused key if it was not in view. Adding 'shouldFocusOnHover':false prevents this scrolling behaviour if an option is hovered over with the mouse, causing unexpected dropdown scrolling

### CODE REVIEW

File: src/components/LinkChooser/LinkChooser.tsx [hunk 1]
```diff
@@ -195,6 +196,16 @@ export const LinkChooser: FC<LinkChooserProps> = ({
...
 
+    useEffect(() => {
+        const { focusedKey } = state.selectionManager;
+        if (focusedKey && popoverRef.current && state.isOpen) {
+            const dropdownItem = popoverRef.current?.querySelector(`[data-key="${focusedKey}"]`);
+            if (dropdownItem) {
+                scrollIntoView(popoverRef.current, dropdownItem as HTMLElement);
+            }
+        }
+    }, [state.selectionManager.focusedKey, state.isOpen]);
```
Comment (NadiaPosch) [inline src/components/LinkChooser/LinkChooser.tsx:207]: You could create variables for the dependencies and then reuse them inside the hook

Comment (SamCreasey) [reply src/components/LinkChooser/LinkChooser.tsx]: Fixed :)

File: src/components/LinkChooser/LinkChooser.tsx [hunk 2]
```diff
@@ -195,6 +196,20 @@ export const LinkChooser: FC<LinkChooserProps> = ({
...
     });
 
+    const {
+        isOpen,
+        selectionManager: { focusedKey },
+    } = state;
+
+    useEffect(() => {
+        if (focusedKey && popoverRef.current && isOpen) {
+            const dropdownItem = popoverRef.current?.querySelector(`[data-key="${focusedKey}"]`);
```
Comment (NadiaPosch) [inline src/components/LinkChooser/LinkChooser.tsx:206]:
```suggestion
            const dropdownItem = popoverRef.current.querySelector(`[data-key="${focusedKey}"]`);
```

Comment (SamCreasey) [reply src/components/LinkChooser/LinkChooser.tsx:206]: This causes a typeError, hence the check afterwards for "if dropdownItem" :)

Comment (SamCreasey) [reply src/components/LinkChooser/LinkChooser.tsx:206]: Oh wait, I'm lying, nevermind!

Comment (SamCreasey) [reply src/components/LinkChooser/LinkChooser.tsx:206]: fixed :)

**Review by NadiaPosch — APPROVED** [review]

**Pull request merged**